apollo/src/datasources/TagResearchDataSource.js

```diff
@@ -143,15 +140,45 @@ class TagResearchDataSource extends DataSource {
     // explicitly ask query ordery by the doc id
     // the orderby usage comes from here
     // https://firebase.google.com/docs/firestore/manage-data/delete-data#collections
+    const docSnap = await this.fixedTagResearchCollectionRef
+      .doc(fixedTagId)
+      .get();
+    // const fixedTagLocation = docSnap.data().locationName;
+    const fixedTagCor = docSnap.data().coordinates;
+
     const query = this.tagResearchDataCollectionRef.where(
       'fixedTagId',
       '==',
       fixedTagId
     );
-
     const snapshot = await query.get();
-    const subTags = [];
-    snapshot.forEach(doc => subTags.push({ ...doc.data(), id: doc.id }));
+    const subTagsTmp = [];
```

Comment:
Suggestion: please try out this approach
```
const listWithDistance = list.map((item) => ({
  data: item,
  distance: calculateDistance(...),
}));

listWithDistance.sort((a, b) => a.distance - b.distance);

return listWithDistance.map((item) => item.data);
```

We can avoid the usage of `delete` with this approach

Reply:
fixed on the commit 36ecd94c8fbbbd2e2b8b05371d67548a7a2dab05